modules/videoio/misc/plugin_ffmpeg/CMakeLists.txt

```diff
@@ -5,7 +5,12 @@ include("${OpenCV_SOURCE_DIR}/cmake/OpenCVPluginStandalone.cmake")
 
 # scan dependencies
 set(WITH_FFMPEG ON)
-set(OPENCV_FFMPEG_SKIP_BUILD_CHECK ON)
+
+if (EMSCRIPTEN)
+  set(OPENCV_FFMPEG_SKIP_BUILD_CHECK OFF)
+else()
+  set(OPENCV_FFMPEG_SKIP_BUILD_CHECK ON)
+endif()
```

Comment:
This file is not used for OpenCV.JS build.
This patch doesn't really work (you probably reuse cached CMake variables instead of clean build).

```
if (EMSCRIPTEN)
  set(OPENCV_FFMPEG_SKIP_BUILD_CHECK OFF)
```

should be "ON" to skip try_compile check for EMSCRIPTEN.

---

Put this code:

```
if(EMSCRIPTEN AND NOT DEFINED OPENCV_FFMPEG_SKIP_BUILD_CHECK)
  set(OPENCV_FFMPEG_SKIP_BUILD_CHECK ON)  # try_compile doesn't work properly with Emscripten
endif()
```

[here](https://github.com/opencv/opencv/blob/4.5.4/modules/videoio/cmake/detect_ffmpeg.cmake#L73).